src/media/local-stream.ts

```diff
@@ -9,12 +10,32 @@ interface LocalStreamEvents {
   [LocalStreamEventNames.ConstraintsChange]: TypedEvent<() => void>;
 }
 
+export type TrackEffect = BaseEffect;
+
+type EffectItem = { name: string; effect: TrackEffect };
+
+/**
+ * Replace an existing track on a media stream for a new track. This method assumes a single track
+ * per stream.
+ *
+ * @param stream - The stream in which the track is to be replaced.
+ * @param track - The track to add to the stream.
+ */
+const replaceTrack = (stream: MediaStream, track: MediaStreamTrack) => {
+  stream.removeTrack(stream.getTracks()[0]);
+  stream.addTrack(track);
+};
+
 /**
  * A stream which originates on the local device.
  */
 abstract class _LocalStream extends Stream {
   [LocalStreamEventNames.ConstraintsChange] = new TypedEvent<() => void>();
 
+  private effects: EffectItem[] = [];
+
+  private loadingEffects: string[] = [];
```

Comment:
Because order isn't important here, I think this could be a map keyed by the effect name, which would make the lookups a bit easier.

Reply:
Updated.